testsuites/syncgateway/functional/tests/test_xattrs.py

```diff
@@ -1212,6 +1213,257 @@ def update_props():
+        assert len(doc_set_ids1) == number_docs_per_client
+        log_info("Docs creation via SDK done")
+        all_docs_via_sg_formatted = [
+            {"id": doc, "rev": "1-"} for doc in doc_set_ids1]
+
+        ct_task = tpe.submit(changestrack.start(timeout=10))
+        log_info("ct_task value {}".format(ct_task))
+        wait_for_changes = tpe.submit(
+            changestrack.wait_until, all_docs_via_sg_formatted, rev_prefix_gen=True)
+
+        if wait_for_changes.result():
+            log_info("Found all docs ...")
+        else:
+            raise Exception(
+                "Could not find all changes in feed for adding docs via SDK before timeout!!")
+
+    with ThreadPoolExecutor(max_workers=5) as tpe:
+        log_info("Updating docs via SDK...")
+
+        # Update docs via SDK
+        sdk_docs = sdk_client.get_multi(doc_set_ids1)
+        assert len(sdk_docs.keys()) == number_docs_per_client
+        for doc_id, val in sdk_docs.items():
+            log_info("Updating: '{}' via SDK".format(doc_id))
+            doc_body = val.value
+            doc_body["updated_by_sdk"] = True
+            sdk_client.upsert(doc_id, doc_body)
+        time.sleep(10)
+        ct_task = tpe.submit(changestrack.start(timeout=15))
+        all_docs_via_sg_formatted = [
+            {"id": doc, "rev": "2-"} for doc in doc_set_ids1]
+
+        wait_for_changes = tpe.submit(
+            changestrack.wait_until, all_docs_via_sg_formatted, rev_prefix_gen=True)
+
+        if wait_for_changes.result():
+            log_info("Found all docs after SDK update ...")
+        else:
+            raise Exception(
+                "Could not find all changes in feed for SDK updated SDK docs before timeout!!")
+
+    # update docs by sync-gateway
+    with ThreadPoolExecutor(max_workers=5) as executor:
+        log_info("Starting updating SDK docs by sync-gateway...")
+        user_docs, errors = sg_client.get_bulk_docs(url=sg_url, db=sg_db, doc_ids=sdk_docs, auth=autosdkuser_session)
+        assert len(errors) == 0
+
+        # Update the 'updates' property
+        for doc in user_docs:
+            doc['updated_by_sg'] = True
+
+        # Add the docs via build_docs
+        sg_docs_update_resp = sg_client.add_bulk_docs(url=sg_url, db=sg_db, docs=user_docs, auth=autosdkuser_session)
+        time.sleep(10)
+        executor.submit(changestrack.start(timeout=10))
+        wait_for_changes = executor.submit(
+            changestrack.wait_until, sg_docs_update_resp)
+
+        if wait_for_changes.result():
+            log_info("Stopping ...")
+            log_info("Found all docs for update docs via sg ...")
+            executor.submit(changestrack.stop)
+        else:
+            executor.submit(changestrack.stop)
+            raise Exception(
+                "Could not find all changes in feed for SG updated SDK docs via sg before timeout!!")
+
+    with ThreadPoolExecutor(max_workers=5) as executor:
```

Comment:
Though they won't be conflicts , just in case if it is , I added unique names for each thread.